bindings/Modules/src/SofaPython3/SofaConstraintSolver/Binding_ConstraintSolver.cpp

```diff
@@ -49,20 +49,59 @@ void moduleAddConstraintSolver(py::module &m)
         return { W_matrix.ptr(), W_matrix.rows(), W_matrix.cols()};
     }, sofapython3::doc::constraintsolver::constraintSolver_W);
 
+    c.def("setW", [](ConstraintSolverImpl& self, py::array_t<SReal> _W)
+    {
+        auto& W_matrix = self.getConstraintProblem()->W;
+
+        auto r = _W.unchecked<2>(); // _W must have ndim = 2; can be non-writeable
+        if (r.shape(0) != W_matrix.rows())
+            throw py::type_error("Invalid row dimension");
+        if (r.shape(1) != W_matrix.cols())
+            throw py::type_error("Invalid col dimension");
+
+        std::memcpy(W_matrix.ptr(), _W.data(), W_matrix.rows() * W_matrix.cols() * sizeof(SReal));
+    }, sofapython3::doc::constraintsolver::constraintSolver_setW);
+
     c.def("lambda_force", [](ConstraintSolverImpl& self) -> Eigen::Map<Eigen::Matrix<SReal, Eigen::Dynamic, 1> >
     {
         assert(self.getConstraintProblem());
         auto& lambda = self.getConstraintProblem()->f;
         return { lambda.ptr(), lambda.size()};
     }, sofapython3::doc::constraintsolver::constraintSolver_lambda);
 
+    c.def("set_lambda_force", [](ConstraintSolverImpl& self, py::array_t<SReal> _lambda)
```

Comment:
Do we really want to have "changing" signature depending on how sofa is compiled ? 
To me this means that a fully valid and working python script made for one "sofa version" will not work on the same sofa version but compiled with a different double/float configuration. I would recommend having here two functions, one for the fast (conversionlesss) path, one for a slow path doing type conversion with a warning saying we are going slow path.